parley/src/shape/cache.rs

```diff
@@ -176,3 +176,27 @@ impl<'a> From<ShapePlanKey<'a>> for ShapePlanId {
         }
     }
 }
+
+/// Cache for charmap mapping indices.
+pub(crate) struct CharmapCache {
+    cache: LruCache<u64, skrifa::charmap::MappingIndex>,
```

Comment:
The key here should include the font index as well. Fonts in ttc files often share the same cmap table but not always so this could potentially lead to incorrect font selection.